src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -2875,6 +2875,899 @@ void M4(bool b)
+                // (37,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 3
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(37, 15),
+                // (46,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 4
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(46, 15)
+                );
+        }
+
+        [Fact]
+        public void EqualsCondAccess_12()
+        {
+            var source = @"
+#nullable enable
+
+class C
+{
+    int? M0(object obj) => null;
+    void M(C? c, object? obj)
+    {
+        int x, y;
+        _ = (c?.M0(x = y = 0) == (int?)obj)
+            ? x.ToString() // 1
+            : y.ToString(); // 2
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (11,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(11, 15),
+                // (12,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(12, 15)
+                );
+        }
+
+        [Fact]
+        public void EqualsCondAccess_13()
+        {
+            var source = @"
+#nullable enable
+
+class C
+{
+    long? M0(object obj) => null;
+    void M(C? c, int i)
+    {
+        int x, y;
+        _ = (c?.M0(x = y = 0) == i)
+            ? x.ToString()
+            : y.ToString(); // 2
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (12,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(12, 15)
+                );
+        }
+
+        [Fact]
+        public void EqualsCondAccess_14()
+        {
+            var source = @"
+#nullable enable
+
+class C
+{
+    long? M0(object obj) => null;
+    void M(C? c, int? i)
+    {
+        int x, y;
+        _ = (c?.M0(x = y = 0) == i)
+            ? x.ToString() // 1
+            : y.ToString(); // 2
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (11,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(11, 15),
+                // (12,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(12, 15)
+                );
+        }
```

Comment:
Does this change include:
```cs
if (e is string s == false) return;
s.ToString(); // ok
```
? From the title I though that should be covered.

Reply:
I expect it does (comparing to bool constants should propagate conditional state). I'll add some tests. Thanks!

Reply:
Done